### DISCUSSION TITLE
Relay nodes

### DISCUSSION THREAD

**Discussion by kehiy** [libp2p/rust-libp2p#4447] [Q&A] [answered]
In this part of `libp2p` docs:

https://docs.libp2p.io/concepts/nat/circuit-relay/#process

The `A` node is already behind a `NAT`, but how does Relay `R` dail it when node `B` tells wants a connection to node `A`?

**Comment by thomaseizinger** [comment, marked as answer]
It doesn't :)

`A` dials the relay and makes a "reservation" and keeps the connection alive.

`R` can then at any point establish a stream to it.

**Reply by kehiy** [reply]
thanksss
